application_sdk/application/metadata_extraction/base.py

```diff
@@ -0,0 +1,144 @@
+
+        # setup workflow client for worker and application server
+        self.workflow_client = get_workflow_client(
+            application_name=self.application_name
+        )
+
+    @observability(logger=logger, metrics=metrics, traces=traces)
+    async def setup_workflow(
+        self,
+        workflow_and_activities_classes: List[
+            Tuple[Type[WorkflowInterface], Type[BaseMetadataExtractionActivities]]
+        ],
+        passthrough_modules: List[str] = [],
+        activity_executor: Optional[ThreadPoolExecutor] = None,
+        max_concurrent_activities: Optional[int] = MAX_CONCURRENT_ACTIVITIES,
+    ):
+        """
+        Set up the workflow client and start the worker for base metadata extraction.
+
+        Args:
+            workflow_and_activities_classes (List[Tuple[Type[WorkflowInterface], Type[BaseMetadataExtractionActivities]]]): List of workflow and activities classes to register. Defaults to [(WorkflowInterface, BaseMetadataExtractionActivities)].
+            passthrough_modules (List[str]): The modules to pass through to the worker. Defaults to [].
+            activity_executor (ThreadPoolExecutor | None): Executor for running activities. Defaults to None.
+            max_concurrent_activities (Optional[int]): Maximum number of concurrent activities. Defaults to MAX_CONCURRENT_ACTIVITIES.
+        """
+        # load the workflow client
+        await self.workflow_client.load()
+
+        workflow_classes = [
+            workflow_class for workflow_class, _ in workflow_and_activities_classes
+        ]
+
+        # Collect all activities from all workflow classes
+        workflow_activities = []
+        for workflow_class, activities_class in workflow_and_activities_classes:
+            workflow_activities.extend(
+                workflow_class.get_activities(
+                    activities_class(
+                        client_class=self.client_class,
+                        handler_class=self.handler_class,
+                        transformer_class=self.transformer_class,
+                    )
+                )
+            )
+
+        self.worker = Worker(
+            workflow_client=self.workflow_client,
+            workflow_classes=workflow_classes,
+            workflow_activities=workflow_activities,
+            passthrough_modules=passthrough_modules,
+            activity_executor=activity_executor,
+            max_concurrent_activities=max_concurrent_activities,
+        )
+
+    @observability(logger=logger, metrics=metrics, traces=traces)
+    async def setup_server(
+        self,
+        workflow_class: Type[WorkflowInterface],
+    ) -> None:
+        """
+        Set up the FastAPI server for the base metadata extraction application.
+
+        Args:
+            workflow_class (Type[WorkflowInterface]): Workflow class to register with the server. Users must provide their own workflow implementation.
+        """
+        if self.workflow_client is None:
```

Comment:
think you meant this
```suggestion
        if self.workflow_client is not None:
```

Reply:
- Copied from the interface : [here](https://github.com/atlanhq/application-sdk/blob/c5222ee0e555491f2ddb959c598748c55467b3d1/application_sdk/application/__init__.py#L164)
- Also in the sql.py : [here](https://github.com/atlanhq/application-sdk/blob/c5222ee0e555491f2ddb959c598748c55467b3d1/application_sdk/application/metadata_extraction/sql.py#L174)

Reply:
Removed the base metadata extraction and made changes to the base application itself.